services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/TableSchema.java

```diff
@@ -237,11 +238,25 @@ default T mapToItem(Map<String, AttributeValue> attributeMap, boolean preserveEm
      * SDK can work with.
      *
      * @param item The modelled Java object to convert into a map of attributes.
-     * @param ignoreNulls If set to true; any null values in the Java object will not be added to the output map.
-     *                    If set to false; null values in the Java object will be added as {@link AttributeValue} of
-     *                    type 'nul' to the output map.
+     * @param configuration configuration options for mapping an object to a map with attribute values
      * @return A map of String to {@link AttributeValue} representing all the modelled attributes in the model object.
      */
+    default Map<String, AttributeValue> itemToMap(T item,
+                                          MappingConfiguration configuration) {
```

Comment:
`MappingConfiguration` seems a bit generic.  `mapToItem` doesn't take a mapping configuration. Maybe just `ItemToMapConfiguration`? 

I just noticed there's another overload down below.
```  
Map<String, AttributeValue> itemToMap(T item, Collection<String> attributes);
```

Are we gonna need to add another overload in the future? `Map<String, AttributeValue> itemToMap(T item, Collection<String> attributes, ItemToMapConfiguration configuration);`

This makes me wonder if we should just add an overload that takes a request wrapper class and returns a response wrapper class to be future proof, something like 

```
ItemToMapResponse itemToMap(ItemToMapRequest request);
```

Reply:
ItemToMap sounds good. I can modify that.